pkg/api/schema/root.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2019-2020 vChain, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package schema
+
+import (
+	"errors"
+	"github.com/codenotary/immudb/pkg/signer"
+	"github.com/golang/protobuf/proto"
+)
+
+func NewRoot() *Root {
+	return &Root{Payload: &RootIndex{}, Signature: &Signature{}}
+}
+
+//CheckSignature
+func (r *Root) CheckSignature() (ok bool, err error) {
```

Comment:
returning error may be enough, nil when signature validates

Reply:
Thanks @jeroiraz, in that case I should parse the error, because inside verify  'unmarshal' could fail. So if an error is found I need to check if there was an unmarshal error or the signature is not valid.